### ISSUE TITLE
Consider boosting the UI font size (boost base or eliminate decreases)

### ISSUE THREAD

**Issue by psobolewskiPhD** [napari/napari#8932]
## 🧰 Task

The base theme font size is set here:
https://github.com/napari/napari/blob/f42b6f278fdaa10a787034115f49dc924cbd1704/src/napari/utils/theme.py#L391

and in many places of our QSS it is *decreased*, which result in very small UI text, especially in the layer controls.

Edit: I think it's the decreasing we use in QSS that is the issue. On macOS 12 is a normal (default) UI font size. So we should audit where we reduce down below that and spot it. This way we won't also boost things that are already fine, like the Welcome.

I commonly hear this complaint from users that the text is very small, particularly when they have a higher res monitor but it's not HiDPI.
While it's easy to change the setting to boost the font, I think it's worth considering whether our default should be larger, as an accessibility issue. As a point of reference: comparing napari to other software I use, it has by far the smallest UI text elements.

Note: changing the base would require looking at all the QSS increase/decrease and consider whether those need to be adjusted as well, particularly the increase.

Also, there is the question of why the theme font size is set in a platform dependent way:
- https://github.com/napari/napari/issues/8931

**Comment by TimMonko** [comment]
I agree that we should increase the font-size generally for the UI, and this is feedback I've been given from others. 
This decision may have been done to prioritize a "simpler/smaller" UI, but we especially have to remember that for folks _unfamiliar_ with the UI that it becomes really hard to parse

Also in my own experience of increasing the font size for myself it does not change the UI size very much. For example a 50% increase in font size changes UI size by a smaller amount

**Comment by psobolewskiPhD** [comment]
smaller != simpler!
If anything it makes things appear more complex, think about lego vs duplo.

**Comment by psobolewskiPhD** [comment]
| base (macOS 12) | +2 (macOS 14) |
| ----------------- | --------------- |
| <img width="414" height="786" alt="Image" src="https://github.com/user-attachments/assets/1e30d3af-cf9c-4cf4-81d8-05a6988a3293" /> | <img width="414" height="786" alt="Image" src="https://github.com/user-attachments/assets/1bc205fd-2610-4e20-a306-8a0f51bb710c" /> |

I don't see any adverse layout issues with this in napari. Going above 14 starts to create some clipping in the title bars, etc.
But, plugins may be different, for example napari-metadata layout results in scrollbars that are annoying (with +2 as above on the right):

<img width="1264" height="809" alt="Image" src="https://github.com/user-attachments/assets/f00fc540-afd0-46a1-81ec-7d49da78830b" />

Note: from an implementation it's not as simple -- IMO -- as just setting the base +2 in theme.py, because of #8931 plus maybe not scaling up further things like command palette and welcome text, which are already much larger than the UI text.

Edit: we may also audit for `decrease` in the QSS to see where font is being reduced below the base. So instead of boosting the base, we may want to not decrease as much in those places. Thinking about it more, I think this is the better solution, because 12 on macOS is the normal/default font sizes most apps use for UI.

**Comment by psobolewskiPhD** [comment]
Here's ripgrep for decrease in our qss:
```
00_base.qss
177-QLineEdit {
178-  background-color: {{ darken(background, 15) }};
179-  color: {{ text }};
180-  min-height: 20px;
181-  padding: 2px;
182-  border-radius: 2px;
183:  font-size: {{ decrease(font_size, 1) }};
--
202-QLabeledSlider > QAbstractSpinBox {
203-  min-width: 10px;
204-  padding: 0px;
205:  font-size: {{ decrease(font_size, 1) }};


02_custom.qss
79-#QtCustomTitleBar > QLabel {
80-  color: {{ text }};
81:  font-size: {{ decrease(font_size, 1) }};
--
176-QtLayerControls > QComboBox, QtLayerControls > QLabel, QtLayerControls, QtPlaneControls > QLabeledSlider > QAbstractSpinBox {
177:  font-size: {{ decrease(font_size, 1) }};
--
194-SliderLabel {
195-  background-color: {{ darken(background, 15) }};
196-  color: {{ text }};
197-  min-height: 16px;
198-  padding: 0px;
199-  border-radius: 2px;
200:  font-size: {{ decrease(font_size, 1) }};
--
215-AutoScaleButtons QPushButton {
216:  font-size: {{ decrease(font_size, 3) }};
--
220-PlaneNormalButtons QPushButton {
221:    font-size: {{ decrease(font_size, 3) }};
--
314-#slice_label {
315:  font-size: {{ decrease(font_size, 1) }};
--
405-NapariQtNotification #source_label {
406-  color: {{ primary }};
407:  font-size: {{ decrease(font_size, 1) }};
--
421-#QtCustomTitleLabel {
422-  color: {{ primary }};
423:  font-size: {{ decrease(font_size, 1) }};
--
548-#small_text {
549-  color: {{ opacity(text, 150) }};
550:  font-size: {{ decrease(font_size, 2) }};
```

**Comment by psobolewskiPhD** [comment]
00_base
- The QLineEdit is used in e.g. in the Points/Shapes face/border color inputs. To my eye there is no reason to have it decreased by 1.
- The QLabeledSlider > QAbstractSpinBox I can't figure out where it's used -- I set it to +10...

02_custom
- #QtCustomTitleBar > QLabel is the font size of the widget titles, e.g. "layer controls" -- removing the `decrease` here is a big boon.
- QtLayerControls > QComboBox, QtLayerControls > QLabel, QtLayerControls, QtPlaneControls > QLabeledSlider > QAbstractSpinBox is basically all of the layer controls. These tend to be the worst offenders for being small, so removing the decrease here is A-OK in my book
- SliderLabel this is the slider label in the layer controls and the contrast widget. If the other controls are being bumped, then this should follow to match.
- AutoScaleButtons QPushButton this is used for the auto contrast buttons for an image layer. Changing this to decrease 1 (instead of 3) doesn't affect the layout and helps a lot with reading it. 
- PlaneNormalButtons QPushButton these the the buttons when you select rendering planes for 3D. As above, changing to decrease 1 helps a ton and does not affect the layout.
- #slice_label this are the numbers on the right side of the dims slider indicating which slice of a stack. Eliminating the decrease makes things match the axis name on the left side of the slider and matches the slider itself better. Pure win.
- NapariQtNotification #source_label is apparently used in notifications, but is None by default and I'm not sure how/where it's set. Can leave it?
- #QtCustomTitleLabel is used in the activity widget you get when you press the button at the bottom right of the viewer status bar. Note that it's also extremely low contrast, using `text` as the color here would be preferable. (Note: the actual activity button is ActivityToggleItem and the text label style is not controlled by qss.)
- #small_text not sure where this is used. I made it +10 and couldn't find it easily, so probably safe to leave it?

**Comment by psobolewskiPhD** [comment]
I think we can close this now? For me the effect is pretty noticeable and the default is pretty legible -- and of course people can boost the base font size in Preferences.
